### ISSUE TITLE
Reproduce Ternary Weight Networks

### ISSUE THREAD

**Issue by ruimashita** [blue-oil/blueoil#24]
Reproduce Ternary Weight Networks

# Background
We want to try to ternary weights instead of 1 bit weights, to get more accuracy. 
I implemented the quantization method of `Ternary Weight Networks`, but I didn't verify the method is correct, yet.

paper: https://arxiv.org/pdf/1605.04711.pdf 
implemented method: https://github.com/blue-oil/blueoil/blob/master/lmnet/lmnet/quantizations/ternary.py#L97-L137

# Task
In order to verify, try to reproduce the experiments following the paper.
They trained `VGG-7` network with `Cifar10` dataset, also trained the `ResNet18` with `ImageNet` dataset.
We have to verify that by using either `VGG-7` or `ResNet18` network trained full precision and ternarized, the both results accuracy is the almost same as the paper.

**Comment by lm-jira** [comment]
I'm gonna try ternary weight quantization on VGG-7 first.
The result will be reported as soon as it finishes.

**Comment by lm-jira** [comment]
I reproduce the VGG7 network from the given paper except that the learning rate is initialized to 0.01 instead of 0.1.
I tried using 0.1 before and it makes the output of network before applying Softmax equal to zero for all classes.

The accuracy for full-precision (gray, orange lines), binarized network (light blue and pink), and ternarized network (dark blue, red) are not significantly different as shown below, 
while the paper shows that binarized network is remarkably worse than other two models. 

<img width="803" alt="screen shot 2018-11-28 at 17 46 06" src="https://user-images.githubusercontent.com/43122237/49139872-2d624a80-f336-11e8-8118-db13d8a5d0a8.png">

**Comment by a-hanamoto** [comment]
@lm-jira Is it done?

**Comment by iizukak** [comment]
Currently, We do not plan to support ternary weight.
I close issue in this time.
